Use AllMetrics for enable & disable lists instead of ignoreWhitelist

eero-t opened the pull request:
Use of AllMetrics instead of ignoreWhitelist, adds CpuUsageMetrics ("cpu") & AppMetrics ("app"), which are already checked by prometheus export code, to metrics that can be disabled/enabled from command line.

That also makes metricSetValue type redundant, as it can now be replaced with MetricSet use.  Removing it requires moving String() & Set() methods, used for options texts and validating given metrics options, to factory.go.

toIncludedMetrics() one-line wrapper is also removed, as inlining container.AllMetrics.Difference() call makes the code intent more explicit, and is about same amount of code.

iwankgb commented on container/factory.go:
```diff
@@ -108,6 +110,30 @@ func (ms MetricSet) Add(mk MetricKind) {
+	return strings.Join(values, ",")
+}
+
+func (ms *MetricSet) Set(value string) error {
+	*ms = MetricSet{}
+	if value == "" {
+		return nil
+	}
+	for _, metric := range strings.Split(value, ",") {
+		if AllMetrics.Has(MetricKind(metric)) {
+			(*ms).Add(MetricKind(metric))
+		} else {
+			return fmt.Errorf("unsupported metric %q specified", metric)
+		}
+	}
+	return nil
+}
```
Can you add a comment to `MetricSet` and state that the struct is not thread safe? And make sure that the struct is not accessed from multiple goroutines in cAdvisor?

eero-t replied:
I added couple of commits clarifying & documenting MericSet thread-safety (removing unused Append() method, changing Add() method to private one and documenting Set() method usage).

iwankgb commented:
/ok-to-test

eero-t commented:
Changes:
- Split toInclude() removal to a separate commit as it's unrelated change touching same lines of code
- Added commit clarifying MetricSet thread-safety (see commit message!)
- Related to that, added commit removing unused MetricSet.Append() method
- Added commit updating the documentation => this PR is now rebased on other PR that fixes documentation for an earlier merged PR

Creatone reviewed (approved):
LGTM
